vars/mtaBuild.groovy

```diff
@@ -1,189 +1,8 @@
-                } else {
-                    echo "[INFO] '${mtaYamlName}' file not found in project sources, but application name provided as parameter - generating '${mtaYamlName}' file."
-                    MtaUtils mtaUtils = new MtaUtils(this)
-                    mtaUtils.generateMtaDescriptorFromPackageJson("package.json", mtaYamlName, applicationName)
-                }
-            } else {
-                echo "[INFO] '${mtaYamlName}' file found in project sources."
-            }
-
-            //[Q]: Why not yaml.dump()? [A]: This reformats the whole file.
-            sh "sed -ie \"s/\\\${timestamp}/`date +%Y%m%d%H%M%S`/g\" \"${mtaYamlName}\""
-
-            def mtaCall
-            def options = []
-
-            String mtarName = configuration.mtarName?.trim()
-            if (!mtarName) {
-                def mtaId = getMtaId(mtaYamlName)
-                mtarName = "${mtaId}.mtar"
-            }
-            options.push("--mtar ${mtarName}")
-
-            switch(configuration.mtaBuildTool) {
-                case 'classic':
-                    // If it is not configured, it is expected on the PATH
-                    def mtaJar = configuration.mtaJarLocation ?: 'mta.jar'
-                    options.push("--build-target=${configuration.buildTarget}")
-                    if (configuration.extension) options.push("--extension=${configuration.extension}")
-                    mtaCall = "java -jar ${mtaJar} ${options.join(' ')} build"
-                    break
-                case 'cloudMbt':
-                    options.push("--platform ${configuration.platform}")
-                    options.push("--target ./")
-                    if (configuration.extension) options.push("--extensions=${configuration.extension}")
-                    mtaCall = "mbt build ${options.join(' ')}"
-                    break
-                default:
-                    error "[ERROR][${STEP_NAME}] MTA build tool '${configuration.mtaBuildTool}' not supported!"
-            }
-
-            echo "[INFO] Executing mta build call: '${mtaCall}'."
-
-            //[Q]: Why extending the path? [A]: To be sure e.g. grunt can be found
-            //[Q]: Why escaping \$PATH ? [A]: We want to extend the PATH variable in e.g. the container and not substituting it with the Jenkins environment when using ${PATH}
-            sh """#!/bin/bash
-            export PATH=./node_modules/.bin:\$PATH
-            $mtaCall
-            """
-
-            script?.commonPipelineEnvironment?.setMtarFilePath("${mtarName}")
-        }
-    }
-}
 
-def String getMtaId(String fileName){
-    def mtaYaml = readYaml file: fileName
-    if (!mtaYaml.ID) {
-        error "Property 'ID' not found in ${fileName} file."
-    }
-    return mtaYaml.ID
+    piperExecuteBin parameters, STEP_NAME, "metadata/${STEP_NAME}.yaml", []
```

Comment:
It would be great if you could also enable the use of the download cache:
```suggestion
    final script = checkScript(this, parameters) ?: this
    parameters = DownloadCacheUtils.injectDownloadCacheInMavenParameters(script, parameters)
    
    piperExecuteBin parameters, STEP_NAME, "metadata/${STEP_NAME}.yaml", []
```